libs/sdk-core/src/models.rs

```diff
@@ -1584,15 +1584,67 @@ impl PaymentPathEdge {
     }
 }
 
+pub(crate) mod sanitize {
```

Comment:
Not sure what is the benefit of a new module and a trait for sanitize. Wouldn't it be simpler if we just add a function to the structures, for example:

```
impl FullReverseSwapInfo {
 fn sanitize(self) -> FullReverseSwapInfo {
            FullReverseSwapInfo {
                preimage: vec![],
                private_key: vec![],
                ..self.clone()
            }
        }
}
```

And then just use it inline where needed?

Reply:
The main benefit is it makes the calling code more readable:

```rust
let reverse_swaps = self
    .persister
    .list_reverse_swaps()
    .map(sanitize_vec_pretty_print)??;
```
vs
```rust
let reverse_swaps = serde_json::to_string_pretty(
    &self
        .persister
        .list_reverse_swaps()?
        .into_iter()
        .map(|rsi| rsi.sanitize())
        .collect::<Vec<FullReverseSwapInfo>>(),
)?;
```

I'm fine with either way, just prefer the first option because it's easier to read.